client/src/main/java/com/defold/extender/client/ExtenderClient.java

```diff
@@ -190,6 +187,7 @@ String queryCache(List<ExtenderResource> sourceResources) throws ExtenderClientE
             if (response.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
                 return EntityUtils.toString(response.getEntity()).replace("\\/", "/");
             } else {
+                EntityUtils.consumeQuietly(response.getEntity());
```

Comment:
First part of the fix. In case of any error on `/query` request we need to consume response body to be sure that all data was read from the socket and connection can be reused in the future.